server/src/db/catalog/partition.rs

```diff
@@ -231,16 +230,28 @@ impl Partition {
         self.tables.values().flat_map(|table| table.chunks.values())
     }
 
-    /// Return an iterator over chunks in this partition that
-    /// may pass the provided predicate
+    /// Return an iterator over chunks in this partition
+    ///
+    /// `table_names` specifies which tables to include
     pub fn filtered_chunks<'a>(
         &'a self,
-        predicate: &'a Predicate,
+        table_names: TableNameFilter<'a>,
     ) -> impl Iterator<Item = &Arc<RwLock<Chunk>>> + 'a {
         self.tables
             .iter()
-            .filter(move |(table_name, _)| predicate.should_include_table(table_name))
-            .flat_map(|(_, table)| table.chunks.values())
+            .filter_map(
+                move |(partition_table_name, partition_table)| match table_names {
+                    TableNameFilter::AllTables => Some(partition_table.chunks.values()),
+                    TableNameFilter::NamedTables(table_names) => {
+                        if table_names.contains(partition_table_name) {
+                            Some(partition_table.chunks.values())
+                        } else {
+                            None
+                        }
+                    }
```

Comment:
I think here you can do this (should pass rustfmt).

```suggestion
                    TableNameFilter::AllTables => Some(partition_table.chunks.values()),
                    TableNameFilter::NamedTables(table_names) => table_names
                        .contains(partition_table_name)
                        .then(|| partition_table.chunks.values()),
```